flink-state-backends/flink-statebackend-changelog/src/main/java/org/apache/flink/state/changelog/ChangelogKeyedStateBackend.java

```diff
@@ -190,6 +194,19 @@
 
     private final ExecutorService asyncOperationsThreadPool;
 
+    /**
+     * Identifies this backend. On restore, when snapshots of multiple backends are present
+     * (downscaling case) this allows to differentiate between state mappings of different backends.
```

Comment:
Actually, upscaling case could also cause this case (such as change parallelism 2 to 3).